Replicator/tests/ReplicatorCollectionTest.cc

```diff
@@ -511,6 +511,93 @@ TEST_CASE_METHOD(ReplicatorCollectionTest, "Multiple Collections Incremental Pus
     validateCollectionCheckpoints(db, db2, 1, "{\"local\":22,\"remote\":24}");
 }
 
+struct Jthread {
+    std::thread thread;
+    Jthread(std::thread&& thread_)
+    : thread(move(thread_))
+    {}
+    ~Jthread() {
+        thread.join();
+    }
+};
+
+TEST_CASE_METHOD(ReplicatorCollectionTest, "Multiple Collections Incremental Revisions", "[Push][Pull]") {
+    addDocs(db, Roses, 2, "db-Roses-");
+    addDocs(db, Tulips, 2, "db-Tulips-");
+    C4Collection* roses = c4db_getCollection(db, Roses, nullptr);
+    C4Collection* tulips = c4db_getCollection(db, Tulips, nullptr);
+    C4Collection* roses2 = c4db_getCollection(db2, Roses, nullptr);
+    C4Collection* tulips2 = c4db_getCollection(db2, Tulips, nullptr);
+
+    SECTION("PUSH") {
+        _expectedDocumentCount = 4;
+        runPushReplication({Roses, Tulips}, {Tulips, Lavenders, Roses});
```

Comment:
The goal of CBL-3501 is to test incremental sync with continuous replicators. So I think this should start a continuous push replicator? The logic could be:

1. Add docs / revs. 
2. Start the thread that adds new revs or docs (Maybe also wait until the replicator is IDLE before start adding new revs to ensue to test collection change observer inside the replicator).
3. Start a continuous push replicator.

Same comment for the other 2 sections.

Reply:
You want the above sync to be Continuous, and wait it until it goes to idle. Then, start the following background thread. Right?

Reply:
Yes or you can do the wait in the background thread as well, which I think it will be easier according how runPushReplication() is written now.